Extension/test/common/internal.ts

```diff
@@ -0,0 +1,91 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
+ * See 'LICENSE' in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+import { AsyncFunc, Func, it } from 'mocha';
+import { is } from '../../src/Utility/System/guards';
+
+export function when(prerequisite: boolean | (() => boolean)): {it: Mocha.TestFunction} {
+    if (is.function(prerequisite)) {
+        prerequisite = prerequisite();
+    }
+    if (prerequisite) {
+        return {it};
+    } else {
+        const skip = (title: string, test: AsyncFunc | Func) => it.skip(`Dynamically skipping test: <<${title}>> - prerequisite not met.`, test);
+        return {it:skip as Mocha.TestFunction};
+    }
+}
+
+import { MessagePort } from 'worker_threads';
+import { collectGarbage } from '../../src/Utility/System/garbageCollector';
+
+function showActiveHandles() {
+    const open = (process as any)._getActiveHandles().filter(
+        (each: any) =>
+            !each.destroyed && // discard handles that claim they are destroyed.
+      !(each.fd === 0) &&  // ignore stdin/stdout/stderr
+      !(each.fd === 1) &&  // ignore stdin/stdout/stderr
+      !(each.fd === 2) &&  // ignore stdin/stdout/stderr
+      !(each instanceof MessagePort) && // ignore worker thread message ports
+      (each.listening) // keep servers that are still listening.
+
+    );
+
+    if (open.length) {
+        console.log('################');
+        console.log('Active Handles: ');
+        console.log('################');
+        console.log(open);
+    // console.log([...getReservations()].join('\n'));
```

Comment:
What is getReservations()? Can you remove this line or add a comment?

Reply:
removed